src/3dm_gx4_node.cpp

```diff
@@ -4,67 +4,210 @@
  *  Created on: Nov 21, 2014
  *      Author: mahisorn
  */
+#include <limits>
+#include <time.h>
 
 #include <ros/ros.h>
 #include <hg_ros_3dm_gx4/3dm_gx4.h>
 #include <sensor_msgs/Imu.h>
+#include <sensor_msgs/MagneticField.h>
+#include <sensor_msgs/NavSatFix.h>
+#include <sensor_msgs/TimeReference.h>
+#include <geometry_msgs/TwistStamped.h>
+#include <geometry_msgs/TwistWithCovarianceStamped.h>
 #include "tf/transform_datatypes.h"
 
 using namespace std;
 
-ros::Publisher g_pub_imu;
-ros::Publisher g_pub_ef_imu;
-sensor_msgs::Imu g_imu;
+ros::Publisher g_pub_raw_imu;
+ros::Publisher g_pub_raw_mag;
+ros::Publisher g_pub_navsat_fix;
+ros::Publisher g_pub_navsat_vel;
+ros::Publisher g_pub_navsat_tf;
+ros::Publisher g_pub_filtered_imu;
+ros::Publisher g_pub_filtered_fix;
+ros::Publisher g_pub_filtered_vel;
+sensor_msgs::Imu g_raw_imu;
+sensor_msgs::MagneticField g_raw_mag;
+sensor_msgs::NavSatFix g_navsat_fix;
+geometry_msgs::TwistStamped g_navsat_vel;
+sensor_msgs::TimeReference g_navsat_tf;
+sensor_msgs::Imu g_filtered_imu;
+sensor_msgs::NavSatFix g_filtered_fix;
+geometry_msgs::TwistWithCovarianceStamped g_filtered_vel;
```

Comment:
What's the reason for all these messages being global? Couldn't they be local in the functions which do the publishing?